connectivity/drivers/nfc/PN512/CMakeLists.txt

```diff
@@ -1,7 +1,34 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-target_include_directories(mbed-os
+# This the library for the PN512 NFC chip
+
+cmake_minimum_required(VERSION 3.18.2 FATAL_ERROR)
+
+# Using relative paths behavior
+if(POLICY CMP0076)
+    cmake_policy(SET CMP0076 NEW)
+endif()
+
+add_library(mbed-os-connectivity-drivers-nfc-pn512 OBJECT)
+
+mbed_set_cpu_core_options(mbed-os-connectivity-drivers-nfc-pn512 ${MBED_TOOLCHAIN})
+mbed_set_toolchain_options(mbed-os-connectivity-drivers-nfc-pn512)
+mbed_set_c_lib(mbed-os-connectivity-drivers-nfc-pn512 ${MBED_C_LIB})
+mbed_set_printf_lib(mbed-os-connectivity-drivers-nfc-pn512 ${MBED_PRINTF_LIB})
+mbed_set_language_standard(mbed-os-connectivity-drivers-nfc-pn512)
+mbed_set_profile_options(mbed-os-connectivity-drivers-nfc-pn512 ${MBED_TOOLCHAIN})
```

Comment:
we shall wrap this to Mbed OS one function to set up entire Mbed Os via one function call. All components will have the same config?

Reply:
I'll create PR shortly to address this and also application configuration

Reply:
this should help https://github.com/ARMmbed/mbed-os/pull/13651

Reply:
Thanks. I will use that PR once it has been approved.